### ISSUE TITLE
Support for Polymer's custom mixins.

### ISSUE THREAD

**Issue by motss** [ben-eb/gulp-cssnano#17]
Encountered an issue when using cssnano on [Polymer's custom mixins](https://www.polymer-project.org/1.0/docs/devguide/styling.html#custom-css-mixins).

Worth to note that there are basically 3 ways to apply a custom mixins in Polymer: 
- `@apply(--mixin-name);`, eg `@apply(--layout-horizontal);`
- `--mixin-name: value`
eg. 
```  
paper-button { /* Polymer's custom element's name */
   --paper-button-ink-color: #009688; /* --mixin-name: #color_code */
}
```
- `--mixin-name: { /* one_or_more_CSS_can_be_applied */ };` eg.
```    
/* 
  custom property that can accept more than one CSS: 
    background-color, border and border-radius. 
*/
--my-toolbar-theme: { 
    background-color: green;
    border-radius: 4px;
    border: 1px solid gray;
};
```
[gulp-minify-css](https://github.com/murphydanger/gulp-minify-css) is going to be deprecated and it leds us here for a better option. However, `gulp-cssnano` does not seem to work well in Polymer's custom mixins and hope that these can be supported so that we can make a switch from the deprecated one without any issue.

**Comment by ben-eb** [comment]
Thanks for the report, let me know if this patch is sufficient.

https://github.com/ben-eb/postcss-minify-selectors/commit/26af0e2eff5363bc922821398145492d9fc2c8fb

**Comment by motss** [comment]
Thanks for the patch. Now it is able to parse the CSS without throwing any error especially for the third type of custom mixins which accepts more than one CSS.

I still have one related question needed to ask if `gulp-cssnano` could cause such problem.

Problem (somehow the custom mixins has been part of the CSS `padding`):
![minify-css-issue](https://cloud.githubusercontent.com/assets/10607759/12236680/30e4c2fe-b8b6-11e5-98bb-600ff7bfc396.png)

Original code with Polymer's custom mixins before minify:
![cssnano-issue](https://cloud.githubusercontent.com/assets/10607759/12236679/30e48a32-b8b6-11e5-8f51-a9d8554bcb22.png)

I have spent some time creating a [simple Polymer element](https://github.com/motss/cssnano-issue-with-polymer) to better the illustrate the issue as it affects some custom mixins but not all which is kind of strange. Hope it helps.

**Comment by ben-eb** [comment]
I'm guessing it's a problem with [rule merging](https://github.com/ben-eb/postcss-merge-rules), so in the meantime you can disable the transform with `{mergeRules: false}`.

**Comment by motss** [comment]
Disabling rule merging doesn't help.

**Comment by ben-eb** [comment]
Can you give me an input and expected output example?

**Comment by motss** [comment]
Okay, I try to make it simple.

Input: 
![cssnano-issue](https://cloud.githubusercontent.com/assets/10607759/12236679/30e48a32-b8b6-11e5-8f51-a9d8554bcb22.png)

Parsed using `gulp-cssnano`:
```
paper-checkbox{display:block;padding:12px;--paper-checkbox-checked-color:#009688;--paper-checkbox-checked-ink-color:#009688;--paper-checkbox-label-color:#009688}
```

Expected output:
![expected-css-output](https://cloud.githubusercontent.com/assets/10607759/12238200/ce6858c2-b8bd-11e5-840e-a9bf77431979.png)

Problem:
![minify-css-issue](https://cloud.githubusercontent.com/assets/10607759/12236680/30e4c2fe-b8b6-11e5-98bb-600ff7bfc396.png)

**Comment by ben-eb** [comment]
I can't see anything wrong with that; all that happened was that the whitespace was removed and the optional closing semicolon was removed. Do Polymer elements require that the final semicolon be present?

**Comment by motss** [comment]
I think you're right. At least 1 space is required for things to work without any issue.


No space at all before a custom property:
```
paper-checkbox{display:block;padding:12px;--paper-checkbox-checked-color:#009688}
```

One space in between the semicolon and the first hyphen of custom property:
```
paper-checkbox{display:block;padding:12px; --paper-checkbox-checked-color:#009688}
```

**Comment by ben-eb** [comment]
Looks like an issue with Polymer itself then. In the meantime you could just disable minification for Polymer mixins. :smile:

**Comment by motss** [comment]
Thank you for your time :+1:

**Comment by robdodson** [comment]
@sorvell does this seem like an easy enough thing to fix?

**Comment by sorvell** [comment]
I recommend filing a [Polymer issue](https://github.com/Polymer/polymer/issues) with a reproduction. I wasn't able to verify a problem given the info here. For example, this seems to function ok: http://jsbin.com/liziba/edit?html,output.

**Comment by motss** [comment]
@sorvell @robdodson Good point though and this allows me to figure out when it can actually happen. 

How about adding one more custom property, say `--paper-checkbox-checkmark-color` after `--paper-checkbox-checked-color`?
[Example to reproduce the issue](http://jsbin.com/fonamisotu/edit?html,output)
![minify-css-reproduce](https://cloud.githubusercontent.com/assets/10607759/12256270/0626318e-b937-11e5-84fe-bda1dcbc3173.png)

And how about prepending an empty space before `--paper-checkbox-checkmark-color`?
[Example to work around the issue](http://jsbin.com/birewadeki/edit?html,output)
![minify-css-reproduce-workaround](https://cloud.githubusercontent.com/assets/10607759/12256451/2022513e-b938-11e5-818e-1be8a2ce9606.png)

Another example with multiple custom properties...
[Something extra](http://jsbin.com/hexojugogi/edit?html,output)

**Comment by TimvdLippe** [comment]
Cross-posting here from https://github.com/PolymerElements/polymer-starter-kit/pull/645#issuecomment-171340876

The parsing of minified CSS should be fixed as of Polymer/polymer#3241. The upcoming release of Polymer includes the fix (or you can use the current master)

@motss can you confirm/deny this has fixed the issue for you?
